apps/test/unit/StudioAppTest.js

```diff
@@ -1,98 +1,131 @@
     });
 
     afterEach(() => {
-      assetsApi.getFiles.restore();
-      listStore.reset.restore();
+      document.body.removeChild(codeWorkspaceDiv);
+      document.body.removeChild(containerDiv);
     });
 
-    it('will pre-populate assets for levels that use assets', () => {
-      studioApp().init({
-        usesAssets: true,
-        enableShowCode: true,
-        containerId: 'foo',
-        level: {
-          editCode: true,
-          codeFunctions: {},
-        },
-        dropletConfig: {
-          blocks: [],
-        },
-        skin: {},
+    beforeEach(() => {
+      stubRedux();
+      registerReducers(commonReducers);
+    });
+    afterEach(restoreRedux);
+
+    describe("the init() method", () => {
+      let files;
+      beforeEach(() => {
+        files = [];
+        sinon.stub(studioApp(), 'configureDom');
+        sinon.stub(assetsApi, 'getFiles').callsFake(cb => cb({files}));
+        sinon.spy(listStore, 'reset');
       });
 
-      expect(assetsApi.getFiles).to.have.been.calledOnce;
-      expect(listStore.reset).to.have.been.calledWith(files);
-    });
+      afterEach(() => {
+        assetsApi.getFiles.restore();
+        listStore.reset.restore();
+      });
 
-    it("will emit an afterInit event", () => {
-      const listener = sinon.spy();
-      studioApp().on('afterInit', listener);
-      studioApp().init({
-        usesAssets: true,
-        enableShowCode: true,
-        containerId: 'foo',
-        level: {
-          editCode: true,
-          codeFunctions: {},
-        },
-        dropletConfig: {
-          blocks: [],
-        },
-        skin: {},
+      it('will pre-populate assets for levels that use assets', () => {
+        studioApp().init({
+          usesAssets: true,
+          enableShowCode: true,
+          containerId: 'foo',
+          level: {
+            editCode: true,
+            codeFunctions: {},
+          },
+          dropletConfig: {
+            blocks: [],
+          },
+          skin: {},
+        });
+
+        expect(assetsApi.getFiles).to.have.been.calledOnce;
+        expect(listStore.reset).to.have.been.calledWith(files);
       });
 
-      expect(listener).to.have.been.calledOnce;
+      it("will emit an afterInit event", () => {
+        const listener = sinon.spy();
+        studioApp().on('afterInit', listener);
+        studioApp().init({
+          usesAssets: true,
+          enableShowCode: true,
+          containerId: 'foo',
+          level: {
+            editCode: true,
+            codeFunctions: {},
+          },
+          dropletConfig: {
+            blocks: [],
+          },
+          skin: {},
+        });
+
+        expect(listener).to.have.been.calledOnce;
+      });
+    });
+  });
+
+  describe('StudioApp.makeFooterMenuItems', () => {
+    beforeEach(() => {
+      sinon.stub(project, 'getUrl');
+    });
+
+    afterEach(() => {
+      project.getUrl.restore();
+    });
+
+    const i18n = {
+      t: key => key
+    };
```

Comment:
Why not just import the `i18n` module and stub the `i18n.t` function the same way you stub `project.getUrl`? Then you can avoid passing in your own i18n module.